app/rust-ffi/src/lib.rs

```diff
@@ -48,18 +48,41 @@ pub fn is_numeric_literal(code: &str) -> bool {
     let parsed = PARSER.with(|parser| parser.parse_block(code));
     let enso_parser::syntax::tree::Variant::BodyBlock(body) = parsed.variant else { return false };
     let [stmt] = &body.statements[..] else { return false };
-    stmt.expression.as_ref().map_or(false, |expr| match &expr.variant {
-        enso_parser::syntax::tree::Variant::Number(_) => true,
-        enso_parser::syntax::tree::Variant::UnaryOprApp(app) =>
-            app.opr.code == "-"
-                && app.rhs.as_ref().map_or(false, |rhs| {
-                    matches!(rhs.variant, enso_parser::syntax::tree::Variant::Number(_))
-                }),
-        _ => false,
+    stmt.expression.as_ref().map_or(false, |expr| {
+        let variant = match &expr.variant {
+            enso_parser::syntax::tree::Variant::ExpressionStatement(expr) => &expr.expression.variant,
+            variant => variant,
+        };
+        match variant {
+            enso_parser::syntax::tree::Variant::Number(_) => true,
+            enso_parser::syntax::tree::Variant::UnaryOprApp(app) =>
+                app.opr.code == "-"
+                    && app.rhs.as_ref().map_or(false, |rhs| {
+                        matches!(rhs.variant, enso_parser::syntax::tree::Variant::Number(_))
+                    }),
+            _ => false,
+        }
```

Comment:
We can return `false` earlier if it's not an `ExpressionStatement`:

```suggestion
    let Some(stmt) = &stmt.expression else { return false };
    let enso_parser::syntax::tree::Variant::ExpressionStatement(stmt) = &stmt.variant else {
        return false;
    };
    match &stmt.expression.variant {
        enso_parser::syntax::tree::Variant::Number(_) => true,
        enso_parser::syntax::tree::Variant::UnaryOprApp(app) =>
            app.opr.code == "-"
                && app.rhs.as_ref().map_or(false, |rhs| {
                    matches!(rhs.variant, enso_parser::syntax::tree::Variant::Number(_))
                }),
        _ => false,
    }
```